Lib/trufont/objects/layoutEngine.py

```diff
@@ -29,17 +29,16 @@ def _layoutEngineOTLTablesRepresentationFactory(layoutEngine):
             compiler = FeatureCompiler(font, otf, kernWriterClass=None)
             compiler.postProcess = lambda: None
             compiler.compile()
-        except:
-            # TODO: handle this in the UI
-            import traceback
-            print(traceback.format_exc(5))
-        else:
             for name in ("GDEF", "GSUB", "GPOS"):
                 if name in otf:
                     table = otf[name].compile(otf)
                     value = hb.Blob.create_for_array(
                         table, HB.MEMORY_MODE_READONLY)
                     ret[name] = value
+        except:
+            # TODO: handle this in the UI
+            import traceback
+            print(traceback.format_exc(5))
```

Comment:
Should we do something there? maybe ret = dict() to drop already collected tables? maybe we don't want

Reply:
I guess so.

BTW, there is also the case when the feature text changes, do you drop the existing tables in this case? If not the we should.

Reply:
See beginSelfFeaturesObservation()

Reply:
I think what's not implemented yet is UFO kerning and the text tool should call the layout engine again when spacing or kerning is changed.

Reply:
For kerning I thought we could use a `get_glyph_h_kerning` callback, tried that now and it does it work and now I remember that HarfBuzz won’t use this calback if the font has a GPOS table. So I guess we need to process the kerning with feaLib after all. Hopefully this will not be a big issue, generating GSUB and GPOS without kerning seems to be pretty fast already.